tensorrt_llm/_torch/attention_backend/sparse/deepseek_v4/deepseek_v4.py

```
@@ -117,3 +152,1296 @@ def get_token_bytes(
+        compress_ratios=sparse_attention_config.compress_ratios,
+        window_size=sparse_attention_config.window_size,
+    )
+
+
+class DeepseekV4TrtllmAttentionMetadata(DSAtrtllmAttentionMetadata):
+    # The set of compress ratios for the layers
+    compress_ratio_set: Set[int]
+    # The set of (compress ratio, attention type) for the layers
+    attention_type_set: Set[Tuple[int, DeepseekV4AttentionType]]
+    # The number of total compressed tokens for each compress ratio
+    num_total_compressed_tokens: Dict[int, int] = {}
```

Ruff (0.15.18)

[warning] Mutable default value for class attribute

(RUF012)